include/sys/util.h

```diff
@@ -97,6 +97,33 @@ constexpr size_t ARRAY_SIZE(T(&)[N]) { return N; }
 #define MIN(a, b) (((a) < (b)) ? (a) : (b))
 #endif
 
+/** @brief Return larger value of two provided expressions.
+ *
+ * Macro ensures that expressions are evaluated only once.
+ *
+ * @note Macro has limited usage compared to the standard macro as it cannot
+ *	 be used to generate constant integers or used outside a function.
+ */
+#define Z_MAX(a, b) ({ \
+		/* random variable names to avoid naming conflict */ \
```

Comment:
Random!?  It's the same every time!

I'll wait for @ceolin to chime in on the need for MISRA tool analysis of the RNG implementation that produced this.

No seriously, this is clever and whimsical and I love it.

Reply:
aren't these declarations totally within the scope of this block anyway? why do we need this?

Reply:
@andrewboie The concern is that either macro argument `a` or `b` may expand to text that was intended to reference a variable with the chosen name in the enclosing scope.

A reasonably common alternative is to just [use a variable that ends with an underscore](https://gcc.gnu.org/onlinedocs/gcc-9.1.0/cpp/Duplication-of-Side-Effects.html#Duplication-of-Side-Effects).  `value_a_` would be a more maintainable choice as it more clearly identifies what it holds and is unlikely to conflict.